apps/mobile/src/libs/components/checkbox-categories-form/checkbox-categories-form.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+	Button,
+	Checkbox,
+	LoaderWrapper,
+	Text,
+	View,
+} from "~/libs/components/components";
+import { BaseColor, DataStatus } from "~/libs/enums/enums";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useFormController,
+	useState,
+} from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import { categoriesSelectedValidationSchema } from "~/packages/categories/categories";
+import { actions as categoriesActions } from "~/slices/quiz/quiz";
+
+import {
+	CATEGORIES_FIELD_NAME,
+	CATEGORIES_FORM_DEFAULT_VALUES,
+	CHECK_ALL_CATEGORIES_NAME,
+	EMPTY_ARRAY_LENGTH,
+} from "./libs/constants/constants";
+import {
+	getAllCategoryIds,
+	getCategoriesSortedByScore,
+	getSelectedCategoryIds,
+} from "./libs/helpers/helpers";
+import { type SelectedCategoryIds } from "./libs/types/types";
+
+type Properties = {
+	onSubmit?: (payload: { categoryIds: SelectedCategoryIds }) => void;
+	submitButtonName: string;
+};
+
+const CheckboxCategoriesForm: React.FC<Properties> = ({
+	submitButtonName,
+}: Properties): JSX.Element => {
+	const [submittedCategoryIds, setSubmittedCategoryIds] =
+		useState<SelectedCategoryIds>(CATEGORIES_FORM_DEFAULT_VALUES);
+
+	const dispatch = useAppDispatch();
+	const { categories, dataStatus } = useAppSelector((state) => state.quiz);
+
+	const { control, errors, handleSubmit, reset, setValue } = useAppForm<{
+		[CATEGORIES_FIELD_NAME]: SelectedCategoryIds;
+	}>({
+		defaultValues: {
+			[CATEGORIES_FIELD_NAME]: CATEGORIES_FORM_DEFAULT_VALUES,
+		},
+		validationSchema: categoriesSelectedValidationSchema,
+	});
+
+	const { field } = useFormController({ control, name: CATEGORIES_FIELD_NAME });
+	const { value } = field;
+
+	const isFormLoading = dataStatus === DataStatus.PENDING;
+	const isAllChecked = value.length === categories.length;
+	const areCategoriesExist = categories.length > EMPTY_ARRAY_LENGTH;
+
+	useEffect(() => {
+		void dispatch(categoriesActions.getScores());
+	}, [dispatch]);
```

Comment:
The form itself shouldn't be responsible for fetching the categories. I think we should dispatch the categories on the exact screen, where the form is used. That means that we also don't need `LoaderWrapper` here and instead of conditionally rendering do:
```
if (!categories.length) {
    return null;
}
```